testsuite/forge/src/backend/local.rs

```diff
@@ -1,16 +1,128 @@
 // Copyright (c) The Diem Core Contributors
 // SPDX-License-Identifier: Apache-2.0
 
-use crate::{AdminInfo, Coffer, Factory, FullNode, NodeId, PublicInfo, Result, Swarm, Validator};
+use crate::{
+    AdminInfo, Coffer, Factory, FullNode, Node, NodeId, PublicInfo, Result, Swarm, Validator,
+};
+use anyhow::ensure;
+use debug_interface::NodeDebugClient;
 use diem_config::config::NodeConfig;
+use diem_genesis_tool::config_builder::FullnodeType;
 use diem_sdk::{
     client::BlockingClient,
     transaction_builder::TransactionFactory,
-    types::{chain_id::ChainId, AccountKey, LocalAccount},
+    types::{chain_id::ChainId, AccountKey, LocalAccount, PeerId},
 };
-use diem_swarm::swarm::DiemSwarm;
+use diem_swarm::swarm::{DiemNode, DiemSwarm, HealthStatus};
+
+struct LocalSwarm {
+    validator_swarm: DiemSwarm,
+    vfn_swarm: DiemSwarm,
+    pfn_swarm: DiemSwarm,
```

Comment:
Later I think we should look at fixing the underlying swarm type itself to be able to support the different types of nodes so we don't have to have multiple "swarms" like this. Course for right now if this works its a quick way to get something off the ground and we can always improve it later.

Reply:
created track ticket https://github.com/diem/diem/issues/8269 for afterward improvement